### PR TITLE
poster

### PR DESCRIPTION
**Pull request by bslc**

### CODE REVIEW

**Comment by kyledemeule** [comment]
@bslc can you merge kyle/master into this branch? Changes have been made since you last pulled and they're showing up in this pull request.

**Comment by bslc** [comment]
```
                                                                                  I am in the gym so not for the next little while.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         From: Kyle DemeuleSent: Saturday, December 5, 2015 10:10 AMTo: kyledemeule/cmpt-726-ripeness-recognitionReply To: kyledemeule/cmpt-726-ripeness-recognitionCc: bslcSubject: Re: [cmpt-726-ripeness-recognition] poster (#10)@bslc can you merge kyle/master into this branch? Changes have been made since you last pulled and they're showing up in this pull request.
```

—Reply to this email directly or view it on GitHub.

File: poster/poster.tex [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+\end{itemize}
+\end{block} %introduction block
+            \vskip1ex
+\begin{alertblock}{Data Collection}
+\begin{itemize}
+\item Took pictures of bananas and non-banana objects.
+\item Lighting, camera (Canon S90) and background were controlled. 
+\item Used 12 unique bananas at various stages to represent the three stages ripeness: unripe, ripe and overripe. 
+\item Non-banana objects: green pepper, apple, tomato, lemon and lime.
```
Comment (kyledemeule) [inline poster/poster.tex]: You have an old dataset, there's also mushrooms, broccoli, potato and pears.

File: poster/poster.tex [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+%define theorem environments
+\setbeamertemplate{theorems}[numbered]
+\newtheorem{proposition}[theorem]{Proposition}
+
+%-----------------------------------------------------------
+% Name and authors of poster/paper/research
+%-----------------------------------------------------------
+
+\title{A Neural Network Approach to Classifying Banana Ripeness }
```
Comment (kyledemeule) [inline poster/poster.tex:100]: Do we want to mention Deep Learning or convolution neural networks in the title? I'm not sure, I actually like it as you wrote it.

File: poster/poster.tex [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+\end{itemize}
+\end{block} %introduction block
+            \vskip1ex
+\begin{alertblock}{Data Collection}
+\begin{itemize}
+\item Took pictures of bananas and non-banana objects.
+\item Lighting, camera (Canon S90) and background were controlled. 
+\item Used 12 unique bananas at various stages to represent the three stages ripeness: unripe, ripe and overripe. 
+\item Non-banana objects: green pepper, apple, tomato, lemon and lime. 
+\item Pictures were resized and cropped to a square.
```
Comment (kyledemeule) [inline poster/poster.tex]: We can also mention the actual pixel dimensions (256 x 256)

File: poster/poster.tex [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+  \centering
+\includegraphics[width=\textwidth]{3_13.jpg}
+\end{subfigure}%
+\vskip 1.ex
+\caption{Upper left: unripe bananas. Upper right: ripe bananas. Lower left: overripe bananas. Lower Right: non-banana objects.}
+\end{figure}
+
+\begin{block}{Methodology}
+ \begin{itemize}
+ \item Used various open source tools for features extraction and classification.
```
Comment (kyledemeule) [inline poster/poster.tex]: Should probably have a one sentence high level description of what we did, e.g:
"Used a pre-trained CNN to extract features from image set and classified those using SVM"
I know it's in the workflow but I'd like to see it written down.

File: poster/poster.tex [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+\end{tabular}
+\end{table}
+
+\vskip1ex
+
+\begin{alertblock}{Results Summary}
+
+\begin{itemize}
+\item (RBF, FC6)  outperformed all other classifiers with 100.0\% accuracy in training and 87.8\% accuracy in testing.
+\item (RBF, FC6) bettered the performance of~\citet{saad2009recognizing} in classifying banana ripeness with significantly larger data set.
```
Comment (kyledemeule) [inline poster/poster.tex]: It didn't though? They had a higher success rate?

File: poster/poster.tex [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+\caption{Architechture of AlexNet.}
+\end{figure}
+\end{alertblock}
+\vskip1ex
+\begin{block}{Results}
+
+\begin{table}[h]
+\caption{Overall percentage of correctly classified objects from training and testing of SVM models with various kernels. Features were obtained from FC6, FC7 and FC8 exits of AlexNet. (Lin = linear, RBF = radial basis function, Sig = sigmoid, Poly = polynomial)}
+\begin{tabular}{|c|c|c|c|c|c|c|c|c|}\hline
+ &  \multicolumn{4}{c|}{Training Results} & \multicolumn{4}{c|}{Testing Results}\\\hline
```
Comment (kyledemeule) [inline poster/poster.tex:311]: I think this is the most important results and should be at the top right of the poster, I dunno we can discuss it today.

File: poster/poster.tex [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+  \end{block} %tools and methods block
+         
+%         \begin{theorem}\label{linearHamiltonianTheorem}
+\vskip1ex
+
+\begin{alertblock}{Features Extraction via AlexNet}
+
+\begin{figure}[h]
+\centering
+\includegraphics[width=\textwidth]{alexnet.png}
```
Comment (kyledemeule) [inline poster/poster.tex:289]: Here's my write up on AlexNet:

Alexnet is a convolutional neural network designed and trained by alex et al to classify images for the ILSVRC-2012 challenge. Alexnet has five convolutional layers, some of which are followed by max-pooling, and then 3 fully connected layers (the first two were trained with dropout, in effect 50% of their weights are 0). A convolutional layer is a neural network layer where each neuron is connected to a small, adjacent portion of neurons on the previous layer. Each neuron on the previous layer is connected to several, adjacent neurons on the next layer. To extract features from our dataset we feed the image through the pre-trained network and extract the internal representation of the image at the last 3 layers (fc6, fc7, and fc8). The internal representation at this layers is either a vector of length 4096 (for fc6 and fc7) or a vector of length 1000 (for fc8). We evaluate classification on the output of each layer.

Let me know if you want to change the format, include other info, clarify something, or change it in anyway.
